beater/beater.go

```diff
@@ -137,7 +135,7 @@ func (bt *beater) Run(b *beat.Beat) error {
 	}
 	defer tracer.Close()
 
-	runServer := bt.runServer
+	runServer := runServer
```

Comment:
For clarity, can't you move L144 up here, and wrap that same `origRunServer` in L166?

Reply:
I'm not too sure what you mean, but I've extracted this tracer-related code to a separate function. Hopefully that will clarify things, let me know.